scripts/test-upgrade.sh

```diff
@@ -105,6 +105,25 @@ cd ${OA_DIR}/playbooks
 ansible galera_all[0] -m shell -a "mysql --verbose -e \"${VLAN_FLAT}\" neutron"
 ansible galera_all[0] -m shell -a "mysql --verbose -e \"${VXLAN}\" neutron"
 
+# CEPH CHANGES
+# These are the instructions to give if there is a failure message on ceph-mon.yml
+if [ $(ansible mons --list-hosts | wc -l) -gt 1 ]; then
+  #Gather facts about all the monitors first, even with bad names
+  #This shouldn't change a thing, it's a regular playbook run.
+  openstack-ansible ${RPCD_DIR}/playbooks/gen-facts.yml
+  for mon in $(ansible mons --list-hosts); do
+    ansible $mon -m command -a "stop ceph-mon id=$mon"
+    ansible $mon -m command -a "ceph mon remove $mon"
+    openstack-ansible lxc-containers-destroy.yml --skip-tags=container-directories --limit $mon
+    openstack-ansible lxc-containers-create.yml --limit $mon
+    # Destroy facts for the current mon, this way it's gonna be properly generated
+    rm -f /etc/openstack_deploy/ansible_facts/$mon
+    openstack-ansible ${RPCD_DIR}/playbooks/ceph-mon.yml --limit $mon
+  done
+  #all the facts are good, they are all part of the cluster. just reconfigure them all
+  openstack-ansible ${RPCD_DIR}/playbooks/ceph-mon.yml
```

Comment:
Yep, as you pointed out this is still necessary.  The first MON that was re-deployed will still reference the other two MONs by their old hostnames, so this ceph-mon.yml run will bring all the ceph.confs in line across the MONs.